pkg/collector/check/check.go

```diff
@@ -209,14 +210,48 @@ func (c *Config) IsTemplate() bool {
 	return false
 }
 
-// GetTemplateVariables returns a slice of raw template variables
-// it found in a config template.
-// FIXME: only extracts from the first instance. Do we need more?
-func (c *Config) GetTemplateVariables() (vars [][]byte) {
-	if len(c.Instances) == 0 {
+// GetTemplateVariablesForInstance returns a slice of raw template variables
+// it found in a config instance template.
+func (c *Config) GetTemplateVariablesForInstance(i int) (vars [][]byte) {
+	if len(c.Instances) < i {
 		return vars
 	}
-	return tplVarRegex.FindAll(c.Instances[0], -1)
+	return tplVarRegex.FindAll(c.Instances[i], -1)
+}
+
+// MergeAdditionalTags merges additional tags to possible existing config tags
+func (c *ConfigData) MergeAdditionalTags(tags []string) {
```

Comment:
this should return an error in case of failure instead of silently not do anything